README.md

```diff
@@ -35,19 +35,21 @@ objectScan(['a.*.f'])({ a: { b: { c: 'd' }, e: { f: 'g' } } });
 - Partial key and index wildcard matching, e.g. `mark*` or `[1*]`
 - Infinite nested matches with `**`
 - Simple or-clause for key and index with `{a,b}` and `[{0,1}]`
+- Input is traversed exactly once and only unique results are returned.
 - Full support for escaping
 - Lots of tests to ensure correctness
 
 ### Options
 
-**Note on Functions:**
-Signature for all functions is `Fn(key, value, { parents, isMatch, needle, needles })`, where:
+**Note on Functions:** Signature for all functions is `Fn(key, value, { parents, isMatch, matches, needles })`, where:
```

Comment:
I'm not sure how I feel about the naming of those parameters... thoughts?

Reply:
```suggestion
**Note on Functions:** Signature for all functions is `Fn(key, value, { parents, isMatch, exactMatches, matches })`, where:
```

Reply:
This pr seems mostly done. Last thing is the naming to sort out...

Reply:
I think I'm ok with leaving it as needles and matches aka `all needles involved` and `all needles that match`. It's not too bad and one can always look it up in the readme as well. Unless we can come up with something better...

Reply:
rename `matches` -> `matchedBy`

Reply:
maybe `needles` -> `checkedBy`?

Reply:
```suggestion
**Note on Functions:** Signature for all functions is `Fn(key, value, { parents, isMatch, matchedBy, traversedBy })`, where:
```